src/main/java/codeu/model/data/MentionList.java

```diff
@@ -0,0 +1,99 @@
+package codeu.model.data;
+
+import codeu.model.store.basic.UserStore;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class MentionList {
+    private Message message;
+    private  UserStore userStore;
+
+    public MentionList(Message message, UserStore userStore){
+        message = this.message;
+        userStore = this.userStore;
+    }
+
+    public List<Mention> getList(){
+        List<Mention> mentions = new ArrayList<>();
+        String content = message.getContent();
+
+        Mention mention = null;
+        boolean inMention = false;
+        int startOfMention = -1;
+        int endOfMention = -1;
+
+        for (int i = 0; i < content.length(); i++) {
+
+            if(content.charAt(i)=='@'){
+                if(inMention){
+                    inMention = false;
+                    break;
+                }
+                else if(i==0 || content.charAt(i-1)==' '){
+                    inMention = true;
+                    startOfMention = i;
+                }
+
+                else if(content.charAt(i) == ' '){
+                    if(inMention){
+                        endOfMention = i;
+
+                        mention = getMention(startOfMention, endOfMention, message);
+                        if(!(mention == null)){
+                            mentions.add(mention);
+                        }
+                        inMention = false;
+                    }
+
+
+                }
+            }
+        }
+
+        return mentions;
+    }
+
+    public Mention getMention(int s, int e, Message m){
```

Comment:
A pointer on class-level API design: I believe you want this to be private, or at least "default" access, not public. This is mostly an internal implementation.